src/js/Controllers/UIController.js

```diff
@@ -388,6 +388,80 @@ class UIController {
+                if(!closepopup_appId) {
+                    return { rpc: RpcFactory.ErrorResponse(rpc, 4, "No active app for interaction") };
+                }
+
+                let closepopup_app = closepopup_state.ui[closepopup_appId]
+                let methodName = rpc.params?.methodName ? rpc.params.methodName :
+                                 (closepopup_app.alert.showAlert && !closepopup_app.alert.isSubtle) ? "UI.Alert" :
+                                 (closepopup_app.alert.showAlert && closepopup_app.alert.isSubtle) ? "UI.SubtleAlert" :
+                                 (closepopup_app.isPerformingInteraction) ? "UI.PerformInteraction":
+                                 //TODO: Add condition for UI Slider interaction
+                                 //TODO: Add condition for UI ScrollableMessage interaction
+                                 //TODO: Add condition for UI PerformAudioPassThru interaction
+                                 null
+
+                if(!methodName){
+                    return { rpc: RpcFactory.ErrorResponse(rpc, 4, "No active interaction to close") };              
+                }
+
+                switch(methodName) {
+                    case "UI.Alert": {
+                        if(!closepopup_app.alert.showAlert || closepopup_app.alert.isSubtle){
+                            return { rpc: RpcFactory.ErrorResponse(rpc, 4, "No active UI.Alert interaction to close") };
+                        }
+                        clearTimeout(this.timers[closepopup_app.alert.msgID])
+                        delete this.timers[closepopup_app.alert.msgID]
+                        this.listener.send(RpcFactory.AlertAbortedResponse(closepopup_app.alert.msgID))
+                        store.dispatch(closeAlert(closepopup_app.alert.msgID, closepopup_appId))
+                        const context = getNextSystemContext();
+                        this.onSystemContext(context, closepopup_appId)
+                        return true;
+                    }
+                    case "UI.SubtleAlert": {
+                        if(!closepopup_app.alert.showAlert || !closepopup_app.alert.isSubtle){
+                            return { rpc: RpcFactory.ErrorResponse(rpc, 4, "No active UI.SubtleAlert interaction to close") };
+                        }
+                        clearTimeout(this.timers[closepopup_app.alert.msgID])
+                        delete this.timers[closepopup_app.alert.msgID]
+                        this.listener.send(RpcFactory.SubtleAlertErrorResponse(closepopup_app.alert.msgID, 
+                            5, 'subtle alert was cancelled'))
+                        store.dispatch(closeAlert(closepopup_app.alert.msgID, closepopup_appId))
+                        const context = getNextSystemContext();
+                        this.onSystemContext(context, closepopup_appId)
+                        return true;
+                    }
+                    case "UI.PerformInteraction": {
```

Comment:
A note for VR.PerformInteraction, it looks like because the Generic HMI sends a SUCCESS response immediately, BOTH interactions are immediately aborted with this change because Core sends a ClosePopUp message. It seems the way around this would be to send an ABORTED response from the Generic HMI (indicating that nothing was selected in the VR interaction), then it will wait for the UI component to respond.

Reply:
Added the ABORTED response for VR.PerformInteraction in https://github.com/smartdevicelink/generic_hmi/pull/398/commits/6a5dc9fcbc40b300dc96b48540c98e27e00f7c61